internal/sdkprovider/service/vpc/aws_org_vpc_peering_connection.go

```diff
@@ -0,0 +1,220 @@
+	},
+	"state": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("State of the peering connection.").PossibleValuesString(organizationvpc.VpcPeeringConnectionStateTypeChoices()...).Build(),
+	},
+}
+
+func ResourceAWSOrgVPCPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Creates and manages an AWS VPC peering connection with an Aiven Organization VPC.",
+		CreateContext: common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionCreate),
+		ReadContext:   common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionRead),
+		DeleteContext: common.WithGenClientDiag(resourceAWSOrgVPCPeeringConnectionDelete),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: schemautil.DefaultResourceTimeouts(),
+
+		Schema: aivenAWSOrgVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceAWSOrgVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	orgID, vpcID, err := schemautil.SplitResourceID2(d.Get("vpc_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var (
+		awsAccountID = d.Get("aws_account_id").(string)
+		awsVPCId     = d.Get("aws_vpc_id").(string)
+		awsRegion    = d.Get("aws_vpc_region").(string)
+
+		req = organizationvpc.OrganizationVpcPeeringConnectionCreateIn{
+			PeerRegion:       util.ToPtr(awsRegion),
+			PeerVpc:          awsVPCId,
+			PeerCloudAccount: awsAccountID,
+		}
+	)
+
+	pCon, err := createPeeringConnection(ctx, orgID, vpcID, client, d, req)
+	if err != nil {
+		return diag.Errorf("Error creating VPC peering connection: %s", err)
+	}
+
+	diags := getDiagnosticsFromState(newOrganizationVPCPeeringState(pCon))
+
+	d.SetId(schemautil.BuildResourceID(orgID, vpcID, awsAccountID, awsVPCId, awsRegion))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		deleteDiags := resourceAzureOrgVPCPeeringConnectionDelete(ctx, d, client)
+		d.SetId("") // Clear the ID after delete
+
+		return append(diags, deleteDiags...)
+	}
+
+	return append(diags, resourceAWSOrgVPCPeeringConnectionRead(ctx, d, client)...)
+}
+
+func resourceAWSOrgVPCPeeringConnectionRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	orgID, orgVpcID, awsAccountID, awsVpcID, awsRegion, err := schemautil.SplitResourceID5(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	vpc, err := client.OrganizationVpcGet(ctx, orgID, orgVpcID)
+	if err != nil {
+		if avngen.IsNotFound(err) {
+			return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+		}
+
+		return diag.Errorf("Error finding VPC: %s", err)
+	}
+
+	pc := lookupAWSPeeringConnection(vpc, awsAccountID, awsVpcID, awsRegion)
+	if pc == nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+	}
+
+	if err = d.Set("vpc_id", schemautil.BuildResourceID(orgID, orgVpcID)); err != nil {
```

Comment:
I see all old vpc do the same story. But it doesn't make sense to me. Why it turns `vpc_id` into a composed field?